Exception/InvalidParameterException.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Exception;
+
+use FOS\RestBundle\Controller\Annotations\Param;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+
+class InvalidParameterException extends BadRequestHttpException
+{
+    private $parameter;
+    private $violations;
+
+    public function __construct(Param $parameter, ConstraintViolationListInterface $violations)
```

Comment:
I added two factory methods instead of adding more arguments to the constructor.